src/refinement/amr_criteria.cpp

```diff
@@ -65,4 +67,39 @@ AmrTag AMRFirstDerivative::operator()(const MeshBlockData<Real> *rc) const {
   return Refinement::FirstDerivative(pmb.get(), q, refine_criteria, derefine_criteria);
 }
 
+
+AMRSecondDerivative::AMRSecondDerivative(ParameterInput *pin, std::string &block_name) {
+  field = pin->GetOrAddString(block_name, "field", "NO FIELD WAS SET");
+  if (field == "NO FIELD WAS SET") {
+    std::cerr << "Error in " << block_name << ": no field set" << std::endl;
+    exit(1);
+  }
```

Comment:
Ideally we should do this with our `PARTHENON_ERROR` machinery. (It's possible the old AMRFirstDerivative was written before that machinery). We need
```C++
#include "utils/error_checking"
```
and then we can do
```C++
PARTHENON_REQUIRE_THROWS(pin->DoesParameterExist(block_name, "field"), "Error in " + block_name + ": no field set.");
field = pin->GetString(block_name, "field");
```

Reply:
Ok, this is easy enough - done.